### DISCUSSION TITLE
Why throw away the Phase spectrum?

### DISCUSSION THREAD

**Discussion by VigneshBaskar** [coqui-ai/TTS#1129] [General]
The output of a TTS system is some form of Magnitude Spectrum (Mel Spectrogram or something else). The Phase spectrum is not at all taken into account for training the TTS system. I understand that both the magnitude spectrum and phase spectrum can be calculated from the Fourier coefficients as follows: 
```python
magnitude = torch.sqrt(real_part_of_fourier_coefficients**2 + imag_part_of_fourier_coefficients**2)
phase = torch.atan2(imag_part_of_fourier_coefficients, real_part_of_fourier_coefficients)
```
So ideally just one of these spectra is sufficient as it already contains all the required Fourier coefficients. But still...

May I request someone to help me with the following questions please:
1. If instead of just training for the magnitude spectrum, if we train for both the magnitude spectrum and the phase spectrum can we naively just use Inverse Fourier Transform for reconstructing the audio signal and ignore this whole phase spectrum guessing game with vocoders?
2.  Does this make sense? 
3.  Is there any research article looking into this direction?

**Comment by erogol** [comment]
I remember reading a couple of papers suggesting that, but it's been a while. I think it's worth trying, especially with one of the normalizing flow models like glow tts as they produce finer outputs.

**Reply by VigneshBaskar** [reply]
Ok Eren. I am training another model currently. I shall try this out once I finish it. Thank you!

**Reply by erogol** [reply]
Awesome!! Let me know how it went. It is really interesting to see the results.

**Reply by VigneshBaskar** [reply]
Sure Eren!
